src/app/modules/timesheet/services/timesheet.service.ts

```diff
@@ -120,14 +120,44 @@ export class TimesheetService {
   }
 
   public openLastTimesheetInLocal(): boolean {
-    const timesheetsOfLocalStorage = this.getTimesheetsLocal();
+    const timesheetsOfLocalStorage = this.getLocalstorageTimesheetsList();
     if (timesheetsOfLocalStorage.length > 0) {
       this.setTimesheet(this._localSaveService.getLocalItem(timesheetsOfLocalStorage[timesheetsOfLocalStorage.length - 1]));
       return true;
     }
     return false;
   }
 
+  public getPresentTimesheet(transferedTimesheet: Timesheet) {
+    const timesheetsOfLocalStorage = this.getLocalstorageTimesheetsList();
+    const localStorageSize = this.getLocalstorageTimesheetsList().length;
+    for (let i = localStorageSize; i >= 0; i--) {
+      const transfertTimesheet: Timesheet = Object.assign({},
+                                                          new Timesheet(),
+                                                          this._localSaveService.getLocalItem(timesheetsOfLocalStorage[i]));
+
+      if (transfertTimesheet.consultant.name === transferedTimesheet.consultant.name &&
+        transfertTimesheet.invoice.provider.name === transferedTimesheet.invoice.provider.name) {
+        return {
+          ...transferedTimesheet,
+          invoice: Object.assign({}, transferedTimesheet.invoice, {
```

Comment:
With the lodash `pick` method you can do this really more simply :
https://lodash.com/docs/4.17.11#pick